pkg/apis/infinispan/v2alpha1/cache_types.go

```diff
@@ -1,24 +1,48 @@
 package v2alpha1
 
 import (
+	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
+// AdminAuth description of the auth info
+type AdminAuth struct {
+	// Secret and key containing the admin username for authentication.
+	Username v1.SecretKeySelector `json:"username,omitempty"`
+	// Secret and key containing the admin password for authentication.
+	Password v1.SecretKeySelector `json:"password,omitempty"`
+}
+
 // CacheSpec defines the desired state of Cache
 type CacheSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "operator-sdk generate k8s" to regenerate code after modifying this file
-	// Add custom validation using kubebuilder tags: https://book-v1.book.kubebuilder.io/beyond_basics/generating_crd.html
+	// Authentication info
+	AdminAuth AdminAuth `json:"adminAuth,omitempty"`
+	// Name of the cluster where to create the cache
+	ClusterName string `json:"clusterName,omitempty"`
+	// Name of the cache to be created. If empty ObjectMeta.Name will be used
+	Name string `json:"name,optional,omitempty"`
+	// Cache template in XML format
+	Template string `json:"template,optional,omitempty"`
```

Comment:
can you pass a cache name only without any xml template and get a local, basic cache? it makes no sense and I'm not sure how it would work. maybe it would just create the cache on the node that gets the request. now that I think about it that's a really bad idea.

so no cache template = no cache gets created?

will there be any sort of check to ensure the template is appropriate? as in my example, if you pass a local cache definition, will the operator create it on the cluster or throw an error?

what about creating caches from the infinispan.org templates?

with the CLI you can do "create cache --template=org.infinispan.DIST_SYNC mycache" 

for the Operator, I'd like to be able to do something like:

```
apiVersion: infinispan.org/v2alpha1
kind: Cache
metadata:
  name: mycache
spec:
  template: org.infinispan.DIST_SYNC
```

Reply:
If no template is provided a cache is created with the same `default` cache template.
I'm not sure if a template name like `org.infinispan.DIST_SYNC` can be passed as REST argument, I'll work on this

Reply:
maybe:

----
POST /rest/v2/caches/mycache?template=org.infinispan.DIST_SYNC
----

Reply:
yeah good, going to implement

Reply:
@oraNod done